scaletest/prebuilds/run.go

```diff
@@ -0,0 +1,365 @@
+
+func (r *Runner) Cleanup(ctx context.Context, _ string, logs io.Writer) error {
+	logs = loadtestutil.NewSyncWriter(logs)
+	logger := slog.Make(sloghuman.Sink(logs)).Leveled(slog.LevelDebug)
+
+	reachedDeletionBarrier := false
+	defer func() {
+		if !reachedDeletionBarrier {
+			r.cfg.DeletionBarrier.Done()
+		}
+	}()
+
+	version, err := r.createTemplateVersion(ctx, r.template.ID, 0, 0)
+	if err != nil {
+		r.cfg.Metrics.AddError(r.template.Name, "create_empty_template_version")
+		return xerrors.Errorf("create empty template version for deletion: %w", err)
+	}
+
+	err = r.client.UpdateActiveTemplateVersion(context.Background(), r.template.ID, codersdk.UpdateActiveTemplateVersion{
+		ID: version.ID,
+	})
+	if err != nil {
+		r.cfg.Metrics.AddError(r.template.Name, "update_active_template_version")
+		return xerrors.Errorf("update active template version to empty for deletion: %w", err)
+	}
+
+	logger.Info(ctx, "waiting for all runners to reach deletion barrier")
+	reachedDeletionBarrier = true
+	r.cfg.DeletionBarrier.Done()
+	r.cfg.DeletionBarrier.Wait()
+	logger.Info(ctx, "all runners reached deletion barrier, proceeding with prebuild deletion")
+
+	err = r.measureDeletion(ctx, logger)
+	if err != nil {
+		return err
+	}
+
+	logger.Info(ctx, "deleting template", slog.F("template_name", r.template.Name))
+
+	err = r.client.DeleteTemplate(ctx, r.template.ID)
+	if err != nil {
+		return xerrors.Errorf("delete template: %w", err)
+	}
+
+	logger.Info(ctx, "template deleted successfully", slog.F("template_name", r.template.Name))
+	return nil
+}
+
+const (
+	PrebuildsTotalLatencyMetric              = "prebuild_total_latency_ms"
+	PrebuildJobCreationLatencyMetric         = "prebuild_job_creation_latency_ms"
+	PrebuildJobAcquiredLatencyMetric         = "prebuild_job_acquired_latency_ms"
+	PrebuildDeletionTotalLatencyMetric       = "prebuild_deletion_total_latency_ms"
+	PrebuildDeletionJobCreationLatencyMetric = "prebuild_deletion_job_creation_latency_ms"
+	PrebuildDeletionJobAcquiredLatencyMetric = "prebuild_deletion_job_acquired_latency_ms"
+)
+
+func (r *Runner) GetMetrics() map[string]any {
+	return map[string]any{
+		PrebuildsTotalLatencyMetric:              r.prebuildTotalLatency.Milliseconds(),
+		PrebuildJobCreationLatencyMetric:         r.prebuildJobCreationLatency.Milliseconds(),
+		PrebuildJobAcquiredLatencyMetric:         r.prebuildJobAcquiredLatency.Milliseconds(),
+		PrebuildDeletionTotalLatencyMetric:       r.prebuildDeletionTotalLatency.Milliseconds(),
+		PrebuildDeletionJobCreationLatencyMetric: r.prebuildDeletionJobCreationLatency.Milliseconds(),
+		PrebuildDeletionJobAcquiredLatencyMetric: r.prebuildDeletionJobAcquiredLatency.Milliseconds(),
+	}
+}
+
+//go:embed tf/main.tf.tpl
+var templateContent string
+
+func TemplateTarData(numPresets, numPresetPrebuilds int) ([]byte, error) {
+	tmpl, err := template.New("prebuilds-template").Parse(templateContent)
+	if err != nil {
+		return nil, err
+	}
+	var result strings.Builder
```

Comment:
I think a normal bytes buffer is fine here, since you never actually care about getting the result as a string.